libipl/p10/ipl0.C

```diff
@@ -344,6 +344,58 @@ static void process_guard_records()
 	}
 }
 
+//@Brief Function will check if the FCO state is set for master proc
+// children or not. If FCO override bit is set for children, need to mark
+// present and functional state to set in devicetree during the boot.
+static bool update_master_proc_fco_state(void)
+{
+	std::array<const char*, 8> mProcChild =
+		{"core", "pauc", "pau", "iohs", "mc", "chiplet", "pec", "fc"};
+
+	struct pdbg_target *proc, *child;
+	uint8_t buf[5];
+
+	pdbg_for_each_class_target("proc", proc) {
+		if(!ipl_is_master_proc(proc))
+			continue;
+
+		for(const char* data : mProcChild) {
+			pdbg_for_each_target(data, proc, child) {
+
+				if (!pdbg_target_get_attribute_packed(child,
+							"ATTR_HWAS_STATE",
+							"41", 1, buf)) {
+					ipl_log(IPL_ERROR, "Attribute [ATTR_HWAS_STATE] read failed\n");
+					return false;
+				}
+
+				//0-1 bits reserved
+				//2nd bit Functional override
+				//3rd bit spec deconfig
+				//4th bit Dump capable
+				//5th bit Functional
+				//6th bit Present
+				//7th bit Deconfig
+				//Checking the FCO bit is set or not. If its set means
+				//functional state of target should be set.
+				if((buf[4] >> 2) & 1) {
+					if (!set_or_clear_state(child, true)) {
+						ipl_log(IPL_ERROR,
+							"Failed to set HWAS state of %s, index %d\n",
+							data, pdbg_target_index(child));
+						return false;
+					}
+				}
+				else {
+					ipl_log(IPL_ERROR, "FCO is not set for index %d\n",pdbg_target_index(child));
```

Comment:
why this is an error ? tyou shouldn't be logging message here , it is going to flood journals.

Reply:
Forgot to remove this. Done.